packages/react/src/hooks/useVisibilityLoader.js

```diff
@@ -86,7 +79,7 @@ const useVisibilityLoader = (ref, options) => {
         observerRef.current = null;
       }
     };
-  }, [isVisible, ref, intersectionObserverOptions, intersectionCallback, hasMoreToLoad]);
+  }, [isVisible, ref, intersectionObserverOptions, hasMoreToLoad, isLoading, onVisible]);
```

Comment:
While we are getting nerdy on the subject, this useEffect will still be called on every render right? Because of the fact that we use lodash merge to create the referenced variables `hasMoreToLoad, isLoading, onVisible, intersectionObserverOptions` and they will be created from a new object each time. Our linting is still gonna complain if we remove or use an empty dependency array so perhaps this is the way to go anyway, but just wanted to clarify.

Reply:
That's a good point. I didn't think about the merge effecting that. Would a `useMemo` handle that?